src/backtracking/hamiltonian_cycle.rs

```diff
@@ -0,0 +1,180 @@
+//! This module provides functionality to find a Hamiltonian cycle in graph.
+//! Source: [Wikipedia](https://en.wikipedia.org/wiki/Hamiltonian_path_problem)
+
+/// Represents a graph with an adjacency matrix.
+struct Graph {
+    /// The adjacency matrix representing the graph.
+    adjacency_matrix: Vec<Vec<u8>>,
```

Comment:
```suggestion
    adjacency_matrix: Vec<Vec<bool>>,
```
And as [before](https://github.com/TheAlgorithms/Rust/pull/730#discussion_r1639932353): what if the input will not be _rectangular_? Maybe it would make sense to introduce a `struct` like `AdjacencyMatrix`.